src/basic/strv.c

```diff
@@ -397,7 +403,16 @@ char *strv_join_prefix(char * const *l, const char *separator, const char *prefi
                 if (prefix)
                         e = stpcpy(e, prefix);
 
-                e = stpcpy(e, *s);
+                bool needs_escaping = unescape_separators && strchr(*s, separator[0]);
+
+                if (needs_escaping)
+                        for (size_t i = 0; (*s)[i]; i++) {
+                                if ((*s)[i] == separator[0])
+                                        *(e++) = '\\';
+                                *(e++) = (*s)[i];
+                        }
+                else
+                        e = stpcpy(e, *s);
         }
 
         *e = 0;
```

Comment:
isn't this a bit ad-hoc? i figure we should escape spaces/tabs/newlines and so on too when used in cryptsetup cmdlines?

i.e. shouldn't this use cescape() or xescape() or some form of it? i.e. first each elements is escaped with these functions and then joined afterwards?

Reply:
I think commas as the only character we will realistically need to escape. The need to escape commas is caused by the mismatch between the kernel names which are passed as commandline arguments like `--cipher=… --foobar` where commas are unambiguous and our format where we squish multiple such arguments into a single word in crypttab: `cipher=…,foobar`. OTOH, whitespace and non-printable characters would be very awkward on the commandline and it's very unlikely that such values will be added. Thus, I think just allowing commas to be escaped is enough to solve the problem. It *is* ad-hoc, but I don't think we need to be a fully generic syntax here: being simple and unambiguous is important too.